src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketBuildStatus.java

```diff
@@ -30,6 +30,36 @@
 
 public class BitbucketBuildStatus {
 
+    /**
+     * Enumeration of possible Bitbucket commit notification states
+     */
+    public enum Status {
+        INPROGRESS {
```

Comment:
How about a `final String` field instead of overrides?

Reply:
Do you mean?
```
public enum Status {
    INPROGRESS("INPROGRESS"),
    FAILED("FAILED"),
       ...
    ;

    private final String status;

    Status(final String status) {
        this.status = status;
    }

    @Override
    public String toString() {
        return status;
    }
}
```

Reply:
Do you mean?
```
public enum Status {
    INPROGRESS("INPROGRESS"),
    FAILED("FAILED"),
       ...
    ;

    private final String status;

    Status(final String status) {
        this.status = status;
    }

    @Override
    public String toString() {
        return status;
    }
}
```

Reply:
Yes, this.